content/child/brave_runtime_features.cc

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/child/brave_runtime_features.h"
+
+#include "brave/components/playlist/buildflags/buildflags.h"
+
+#if BUILDFLAG(ENABLE_PLAYLIST)
+#include "base/feature_list.h"
+#include "brave/components/playlist/features.h"
+#include "third_party/blink/public/platform/web_runtime_features.h"
+#endif
+
+namespace content {
+
+void BraveSetRuntimeFeaturesDefaultsAndUpdateFromArgs(
+    const base::CommandLine& command_line) {
+#if BUILDFLAG(ENABLE_PLAYLIST)
+  // When MediaSource API is enabled, we'll get "blob: " source url. We can't
+  // download media files from it, so we disable this API when playlist is
+  // eanbled.
+  blink::WebRuntimeFeatures::EnableFeatureFromString(
+      "MediaSourceStable",
+      !base::FeatureList::IsEnabled(playlist::features::kPlaylist));
+#endif
```

Comment:
@bridiver @goodov This is the new way to MediaSource API without JS injection. What do you think?

Reply:
this approach is better, but it's still global and there's no way to turn it off other than a feature flag. It's possible that a lot of websites (conference call, streaming services, internal corporate services, etc.) may use this API to work properly. Do we want to break them?

> We're going to support a global preference or content setting for this behavior. But for now we don't have any specific plan.

Do we really need this change for all playlists to work? If it's required for few websites then why not just add it for these specific websites?

I see few options:
1. add an "allow-list" or per-website setting in this PR. 
2. postpone this exact change and make it as a standalone PR to disable this JS endpoint in the right (non-global) way.
3. hardcode (for now?) websites that need `MediaSource` hiding.